### PR TITLE
Bump aws-janitor image tag, and use date-sha format

### PR DESCRIPTION
**Pull request by krzyzacy**
picks up https://github.com/kubernetes/test-infra/pull/7740

/assign @justinsb @chrislovecnm @BenTheElder

### CODE REVIEW

File: maintenance/aws-janitor/Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 IMAGE ?= gcr.io/k8s-test-infra-aws/aws-janitor
-VERSION ?= 0.4
+VERSION = $(shell date +v%Y%m%d)-$(shell git describe --tags --always --dirty)
```
Comment (justinsb) [inline maintenance/aws-janitor/Makefile]: I'd probably stick with `?=` so we can specify a static version, but in practice we don't want to IMO

**Comment by krzyzacy** [comment]
/hold
I can keep that

**Comment by krzyzacy** [comment]
/hold cancel
whoops

**Pull request merged**